gradio/events.py

```diff
@@ -778,11 +792,31 @@ class Events:
         doc="This listener is triggered when the user streams the {{ component }}.",
         connection="stream",
         show_progress="minimal",
+        event_specific_args=[
+            {
+                "name": "stream_every",
+                "type": "float = 0.5",
+                "doc": "The latency (in seconds) at which stream chunks are sent to the backend. Defaults to 0.5 seconds. Parameter only used for the `.stream()` event.",
+            },
+            {
+                "name": "time_limit",
+                "type": "float | None = None",
+                "doc": "The time limit for the function to run. Parameter only used for the `.stream()` event.",
+                "component_prop": "false",
+            },
+        ],
     )
     like = EventListener(
         "like",
         config_data=lambda: {"likeable": False},
         callback=lambda block: setattr(block, "likeable", True),
+        event_specific_args=[
```

Comment:
event_specific_args is used in two places:
1. In the `.pyi` template so that `like_user_message` is only shown in the IDE hint of the like event

<img width="866" alt="Screenshot 2024-09-11 at 12 23 29 PM" src="https://github.com/user-attachments/assets/2378fe4f-3107-4f54-ad90-9dfcb33ea732">

2. In the `init.ts` file so that we know which props must be set from the events